dissect/target/plugins/scrape/qfind.py

```diff
@@ -25,27 +36,32 @@ def check_compatible(self) -> None:
     @arg("-n", "--needles", type=str, nargs="*", metavar="NEEDLES", help="needles to search for")
     @arg("-nf", "--needle-file", type=Path, help="file containing the needles to search for")
     @arg("-e", "--encoding", type=str, help="encode text needles with these comma separated encodings")
-    @arg("--no-hex-decode", action="store_true", help="do not automatically add decoded hex needles (only in raw mode)")
     @arg("-R", "--raw", action="store_true", help="show raw hex dumps instead of post-processed string output")
+    @arg("-r", "--regex", action="store_true", help="parse needles as regexes")
+    @arg("--no-hex-decode", action="store_true", help="do not automatically add decoded hex needles (only in raw mode)")
     @arg("-i", "--ignore-case", action="store_true", help="case insensitive search")
     @arg("--allow-non-ascii", action="store_true", help="allow non-ASCII characters in the output")
     @arg("-u", "--unique", action="store_true", help="only show unique string hits (does not apply to raw output)")
     @arg("-W", "--window", type=int, default=256, help="maximum window size in bytes for context around each hit")
-    @export(output="none")
+    @export(record=QFindHitRecord)  # can be QFindHitRecord or none output depending on invocation
     def qfind(
         self,
         needles: list[str] | None = None,
         needle_file: Path | None = None,
         encoding: str = "",
         no_hex_decode: bool = False,
         raw: bool = False,
+        regex: bool = False,
         ignore_case: bool = False,
         allow_non_ascii: bool = False,
         unique: bool = False,
         window: int = 256,
-    ) -> None:
+    ) -> Iterator[QFindHitRecord] | None:
         """Find a needle in a haystack.
 
+        Needles starting with a ``#`` are ignored in needle files.
+        Hex encode such needles to include them in the needle stack.
```

Comment:
Took me a minute to understand that you meant to hex encode needles starting with `#`, can you rephrase?